### PR TITLE
Select Stocks by Volume Increases

### LINKED ISSUE #586: Select Stocks by Volume Increases
# Aim

- Script to determine and choose stocks whose volume increased over a given amount of time

# Details

-The user can provide the dataset, number of days for comparison and also how many times the volume must be increased. Then after taking the inputs the script will check and give the results of the stocks.

Updates: Made it real-time, inc points

### PR DESCRIPTION
**Pull request by Amit366**
# Description

Script to determine and choose stocks whose volume increased over a given amount of time

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: Select Stocks by volume Increase/script.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+for stock in df[stocks]:
+    stock = stock.upper()
+    if '^' in stock:
+        pass
+    else:
+        try:
+            stock_info = yf.Ticker(stock)
+            
+            hist = stock_info.history(period=days)
+            previous_averaged_volume = hist['Volume'].iloc[1:period-1:1].mean()
```
Comment (antrikshmisri) [inline Select Stocks by volume Increase/script.py:22]: Remove the unused variable `period` to pass the tests

**Review by antrikshmisri — CHANGES_REQUESTED** [review]

File: Select Stocks by volume Increase/script.py (hunk 1 above)
Comment (Amit366) [reply Select Stocks by volume Increase/script.py:22]: Period here is the number of days for comparison.

**Comment by Amit366** [comment]
@antrikshmisri now is it fine?

**Review by antrikshmisri — APPROVED** [review]
@Amit366 LGTM 👍🏻

**Review by kaustubhgupta — COMMENTED** [review]
From which url we can get the latest stocks csv?

**Comment by Amit366** [comment]
I have given a reference CSV file

**Comment by kaustubhgupta** [comment]
> I have given a reference CSV file @kaustubhgupta

That I have seen, but as you suggested it will be real-time, what is real-time in this?

**Comment by Amit366** [comment]
I haven't said it's real time. The user will provide the data. @kaustubhgupta

**Comment by Amit366** [comment]
I have asked yesterday that I can make another one that is real time, but it was said first try this out.

**Comment by kaustubhgupta** [comment]
> I have asked yesterday that I can make another one that is real time, but it was said first try this out.

Okay then make it real-time and I will increase the points for this issue

**Comment by Amit366** [comment]
@kaustubhgupta yesterday I asked but it was said do this and then real time.

**Comment by Amit366** [comment]
Now I have to create totally a new project.

**Comment by kaustubhgupta** [comment]
@Amit366 we don't want your script to be static and we don't want you to make separate projects. Making this existing code real-time is just a few lines of code change but if we merge this and assign a new project for real-time, it won't make any sense.

**Comment by Amit366** [comment]
Ok then

**Comment by antrikshmisri** [comment]
> @kaustubhgupta yesterday I asked but it was said do this and then real time.

@Amit366  It was decided to first implement a user input based script then update it to work with real time data

**Comment by Amit366** [comment]
@kaustubhgupta review now

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Did you ran this script before pushing it?

- You are taking "int" choice from the user but there is no int function to convert the string input from the terminal and therefore, ch==1 will never be executed. 
- In option two, you made it only else which means I can give any character other than 2 to execute the else part.
- Shift the import statements of else part to the top.
- In else part, the API_KEY is exposed in public which is not supposed to happen. Store it in a file and then retrieve it. 
- Reframe this "period=input("Interval for comparison with units: ")" as period=input("Interval for comparison with units (ex 5min): ")
- Remove line 48 (print statement), 50 (i=1)

**Comment by Amit366** [comment]
@kaustubhgupta I did run it and it worked fine.

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta I did run it and it worked fine.

Did you try running for the 1st option, it's not possible!

**Comment by Amit366** [comment]
@kaustubhgupta I have updated the changes

File: Select Stocks by volume Increase/script.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+print("2) Real time comparison of stocks")
+ch = int(input("Enter choice: "))
+if(ch==1):
+  data = input("Enter file path: ")
+  df = pd.read_csv(data)
+
+  stocks = input("Enter column heading of stocks: ")
+  days = input("Enter number of days for comparision: ")
+  timesincrease = input("Enter number of times increase in volume required: ")
+  print(df[stocks])
```
Comment (kaustubhgupta) [inline Select Stocks by volume Increase/script.py:17]: Remove this print statement

File: Select Stocks by volume Increase/script.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  api_key = os.environ['api_key']
+  stocks=input("Name of the stock: ")
+  
+  ts = TimeSeries(key=api_key, output_format='pandas')
+  data, meta_data = ts.get_intraday(symbol=stocks, interval = '5min', outputsize = "full")
+
+  close_data = data['4. close']
+  percentage_change = close_data.pct_change()
+
+  print(percentage_change)
```
Comment (kaustubhgupta) [inline Select Stocks by volume Increase/script.py:51]:
```suggestion
  print("Percentage Change: " + str(percentage_change))
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Before I approve, minor enhancements:

**Comment by Amit366** [comment]
@kaustubhgupta updated the changes

**Review by kaustubhgupta — APPROVED** [review]

**Comment by Amit366** [comment]
@kaustubhgupta is there any other issue which is open??

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta is there any other issue which is open??

You can filter issues with "up-for-grab" label: https://github.com/avinashkranjan/Amazing-Python-Scripts/issues?q=is%3Aopen+is%3Aissue+label%3A%22up+for+grab%22

**Review by antrikshmisri — APPROVED** [review]
LGTM 👍🏻

**Review by santushtisharma10 — APPROVED** [review]
LGTM

**Review by avinashkranjan — APPROVED** [review]

**Pull request merged**